### PR TITLE
More migrations of Version for index created version to IndexVersion

### PR DESCRIPTION
**Pull request by thecoop**
More migrations of index created version from `Version` to `IndexVersion`. The majority of these are straight swaps, non-trivial changes are commented

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/index/analysis/PreConfiguredTokenFilterTests.java [hunk 1]
```diff
@@ -132,8 +132,8 @@ public boolean incrementToken() {
         );
         assertSame(tff_v1_1, tff_v1_2);
 
-        byte major = VersionUtils.getFirstVersion().major;
-        Version version2 = Version.fromString(major - 1 + ".0.0");
+        int beforeAll = IndexVersionUtils.getFirstVersion().id() - 100;
+        IndexVersion version2 = IndexVersion.fromId((beforeAll / 1_000_000) * 1_000_000);
```
Comment (thecoop) [inline server/src/test/java/org/elasticsearch/index/analysis/PreConfiguredTokenFilterTests.java:136]: This should be enough to capture 'version definitely before the first version', as we bump by 1,000,000 every major release

**Review by williamrandolph — CHANGES_REQUESTED** [review]
Looks good, though I have one small suggestion for clarity/maintainability.

File: server/src/test/java/org/elasticsearch/index/analysis/PreConfiguredTokenFilterTests.java (hunk 1 above)
Comment (thecoop) [reply server/src/test/java/org/elasticsearch/index/analysis/PreConfiguredTokenFilterTests.java]: Good idea - spotted an existing method I could use

**Review by williamrandolph — APPROVED** [review]
LGTM!

**Pull request merged**